src/Integration.Vsix/SonarLintTracker/SinkManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.Shell.TableManager;
+using System;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    /// <summary>
+    /// Every consumer of data from an <see cref="ITableDataSource"/> provides an <see cref="ITableDataSink"/> to record the changes. We give the consumer
+    /// an IDisposable (this object) that they hang on to as long as they are interested in our data (and they Dispose() of it when they are done).
+    /// </summary>
+    internal class SinkManager : IDisposable
+    {
+        private readonly TaggerProvider taggerProvider;
+        private readonly ITableDataSink sink;
+
+        internal SinkManager(TaggerProvider taggerProvider, ITableDataSink sink)
```

Comment:
This is fine, but in general if the class is internal, the public constructor has internal access too, hence this could be changed to the following without behavior change:

    public SinkManager(...)

in other words, the members inherit the class accessibility if it is lower (I would guess this is the same in Java too)